ui/src/app/machines/views/MachineDetails/MachineNetwork/EditPhysicalForm/EditPhysicalFields/EditPhysicalFields.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { mount } from "enzyme";
+import { Formik } from "formik";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import EditPhysicalFields from "./EditPhysicalFields";
+
+import type { NetworkInterface } from "app/store/machine/types";
+import type { RootState } from "app/store/root/types";
+import {
+  fabric as fabricFactory,
+  fabricState as fabricStateFactory,
+  machineDetails as machineDetailsFactory,
+  machineInterface as machineInterfaceFactory,
+  machineState as machineStateFactory,
+  machineStatus as machineStatusFactory,
+  machineStatuses as machineStatusesFactory,
+  rootState as rootStateFactory,
+  subnet as subnetFactory,
+  subnetState as subnetStateFactory,
+  vlan as vlanFactory,
+  vlanState as vlanStateFactory,
+} from "testing/factories";
+import { waitForComponentToPaint } from "testing/utils";
+
+const mockStore = configureStore();
+
+describe("EditPhysicalFields", () => {
+  let nic: NetworkInterface;
+  let state: RootState;
+
+  beforeEach(() => {
+    nic = machineInterfaceFactory({
+      id: 1,
+    });
+    state = rootStateFactory({
+      fabric: fabricStateFactory({
+        items: [fabricFactory({}), fabricFactory()],
+        loaded: true,
+      }),
+      machine: machineStateFactory({
+        items: [
+          machineDetailsFactory({
+            interfaces: [nic],
+            system_id: "abc123",
+          }),
+        ],
+        statuses: machineStatusesFactory({
+          abc123: machineStatusFactory(),
+        }),
+      }),
+      subnet: subnetStateFactory({
+        items: [subnetFactory(), subnetFactory()],
+        loaded: true,
+      }),
+      vlan: vlanStateFactory({
+        items: [vlanFactory(), vlanFactory()],
+        loaded: true,
+      }),
+    });
+  });
+
+  it("does not allow the link speed to be higher than the interface speed", async () => {
```

Comment:
This should probably read "shows a warning if link speed is higher than interface speed"

Reply:
Thanks, done.